packages/telemetry/src/telemetryReporter.ts

```diff
@@ -47,15 +49,25 @@ export default class TelemetryReporter extends AsyncCreatable<TelemetryOptions>
    * @param attributes {Attributes} - map of properties to publish alongside the event.
    */
   public sendTelemetryEvent(eventName: string, attributes: Attributes = {}): void {
-    if (this.appInsightsClient) {
-      const name = `${this.options.project}/${eventName}`;
-      const { properties, measurements } = buildPropertiesAndMeasurements(attributes);
-      this.logger.debug(`Sending telemetry event: ${name}`);
-      this.appInsightsClient.trackEvent({ name, properties, measurements });
-      this.appInsightsClient.flush();
-    } else {
-      this.logger.warn('Failed to send telemetry event because appInsightsClient does not exist');
-      throw Error('Failed to send telemetry event because appInsightsClient does not exist');
+    if (this.isSfdxTelemetryEnabled()) {
```

Comment:
This is personal preference but I try to avoid nesting `if`s. I 'd suggest doing something like this to clean it up:

```typescript
public sendTelemetryEvent(eventName: string, attributes: Attributes = {}): void {
    if (!this.isSfdxTelemetryEnabled) {
        return;
    }
    // etc...
}
```

Reply:
Yeah sure I like that better also. Thanks